### PR TITLE
Update CHANGELOG.md for Dart 2.4.0 stable

### PR DESCRIPTION
**Pull request by aadilmaan**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -139,10 +77,46 @@ Updated the linter to `0.1.89`, which includes the following changes:
...
+
+[35097]: https://github.com/dart-lang/sdk/issues/35097
+
+### Dart VM
+
+### Tools
+
+#### Linter
+
+The Linter was updated to `0.1.91`, which includes the following changes:
```
Comment (Stargator) [inline CHANGELOG.md]: If this is going to have linter 0.1.91, then it's missing a mention of the new lint `prefer_double_quotes` ([link](https://github.com/dart-lang/linter/commit/fd1ff984e54bc8c93767effe999f65bd970793a8))

**Comment by kevmoo** [comment]
See https://dart-review.googlesource.com/c/sdk/+/105727

File: CHANGELOG.md (hunk 1 above)
Comment (athomas) [reply CHANGELOG.md]: * New lint: `prefer_double_quotes` is in the list now, so this looks done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -149,10 +95,50 @@ Updated the linter to `0.1.89`, which includes the following changes:
...
+
+* fixed missed cases in `prefer_const_constructors`
+* fixed `prefer_initializing_formals` to no longer suggest API breaking changes
+* updated `omit_local_variable_types` to allow explicit `dynamic`s
+* (internal) migration from deprecated analyzer APIs
```
Comment (athomas) [inline CHANGELOG.md]: I don't think we need to expose internal linter changes in the SDK changelog.

Comment (aadilmaan) [reply CHANGELOG.md]: Updated.

File: CHANGELOG.md (hunk 1 above)
Comment (Stargator) [reply CHANGELOG.md]: @athomas Looks like it's mentioned twice now.

**Pull request closed without merging**